src/services/user/registration.js

```diff
@@ -142,16 +154,20 @@ const registerUser = function register(data, params, app) {
 		return Promise.resolve();
 	})
 		.then(() => {
-			const userMail = data.parent_email || data.student_email || data.email;
-			const pinInput = data.pin;
+			const email = data.parent_email || data.student_email || data.email;
+			const { pin } = data;
 			return app.service('registrationPins').find({
-				query: { pin: pinInput, email: userMail, verified: false },
-			}).then((check) => {
+				query: { pin, email },
+			}).then((result) => {
 				// check pin
-				if (!(check.data && check.data.length > 0 && check.data[0].pin === pinInput)) {
-					return Promise.reject('Ungültige Pin, bitte überprüfe die Eingabe.');
+				if (result.data.length !== 1 || result.data[0].verified !== true) {
+					return Promise.reject(new Error('Der eingegebene Code konnte leider nicht verfiziert werden. Versuch es doch noch einmal.'));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [This line has a length of 145. Maximum allowed is 120.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5203912)